ShopApp/ShopApp/Controller/ProductDetailsViewController.swift

```diff
@@ -1,300 +1,259 @@
+         if let pageViewController = destination as? ProductImagesPageViewController {
+             walkthroughPageViewController = pageViewController
+             walkthroughPageViewController?.walkthroughDelegate = self
+             pageViewController.setProduct(product: self.product)
+         }
+
+         if let commentViewontroller = destination as? AddCommentViewController {
+             guard let email = Auth.auth().currentUser?.email as? String else { return }
+             commentViewontroller.setRoot(root: "comments/\(product.getName())/\(email.replacingOccurrences(of: ".", with: " "))")
+         }
+     }
+
+     // MARK: - CloseAction for close button
+     @IBAction private func closeAction(sender: UIButton) {
+         dismiss(animated: true, completion: nil)
+     }
+
+     // MARK: - LikeAction for like button
+     @IBAction private func likeAction(sender: UIButton) {
+         self.favoritesFunction { action in
+             switch action {
+             case "added":
+                 print("ADDED")
+                 return
+             case "removed":
+                 print("REMOVED")
+                 return
+             default:
+                 return
+             }
+         }
+     }
+
+     // MARK: - Add to cart action
+     @IBAction private func addToCart(sender: UIButton) {
+         if self.sizeSelected == "" {
+             let allFieldRequired = UIAlertController(title: "Message",
+                                                      message: "First you have to choose a size!",
+                                                      preferredStyle: .alert)
+             allFieldRequired.addAction(UIAlertAction(title: String(localized: "Ok", comment: "Ok"), style: .cancel, handler: nil))
+             self.present(allFieldRequired, animated: true, completion: nil)
+         } else {
+             self.addToCartAction(size: self.sizeSelected) { messagge in
+                 let allFieldRequired = UIAlertController(title: "Message",
+                                                          message: messagge,
+                                                          preferredStyle: .alert)
+                 allFieldRequired.addAction(UIAlertAction(title: String(localized: "Ok", comment: "Ok"), style: .cancel, handler: nil))
+                 self.present(allFieldRequired, animated: true, completion: nil)
+             }
+         }
+     }
+
+     // MARK: - Size button select actoin
+     @IBAction private func sizeSelect(sender: UIButton) {
+         guard sender.tintColor != UIColor.systemGray3 else {
+             return
+         }
+
+         for i in 0...5 {
+             sizeButtons[i].backgroundColor = .systemGray6
+         }
+
+         sender.backgroundColor = .systemGray3
+         guard let name = sender.titleLabel?.text as? String else {
+             return
+         }
+
+         self.sizeSelected = name
+     }
+
+     func favoritesFunction(complition: @escaping (String) -> Void) {
```

Comment:
Rename the function to a more intuitive name, like favoritesAction or favoritesToggle. Also there is a typo, "completion"